hudi-spark/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -16,8 +16,10 @@
  * limitations under the License.
  */
 
-package org.apache.hudi;
+package org.apache.hudi.keygen;
```

Comment:
This is a backwards incompatible change. Users would have custom key generators using configuration : DataSourceWriteOptions.KEYGENERATOR_CLASS_OPT_KEY()

It makes sense to move to separate package but we need to call out the change in release notes . Please open a tracking ticket to update release notes for this.

Reply:
+1 @bschell